squidb/src/com/yahoo/squidb/data/SquidDatabase.java

```diff
@@ -58,24 +68,39 @@
  * <p>
  * Methods that use String arrays for where clause arguments ({@link #update(String, ContentValues, String, String[])
  * update}, {@link #updateWithOnConflict(String, ContentValues, String, String[], int) updateWithOnConflict}, and
- * {@link #delete(String, String, String[]) delete}) are wrappers around Android's {@link SQLiteDatabase} methods, and
- * so are left intact. However, Android's default behavior of binding all arguments as strings can have unexpected
- * bugs, particularly when working with SQLite functions. For example:
+ * {@link #delete(String, String, String[]) delete}) are wrappers around Android's {@link SQLiteDatabase} methods.
+ * However, Android's default behavior of binding all arguments as strings can have unexpected bugs, particularly when
+ * working with SQLite functions. For example:
  *
  * <pre>
  * select * from t where _id = '1'; // Returns the first row
  * select * from t where abs(_id) = '1'; // Always returns empty set
  * </pre>
  *
- * {@link DatabaseDao} contains workarounds for this behavior, so all users are encouraged to use DatabaseDao for these
- * operations. If you choose to call these methods directly on AbstractDatabase instead, you have been warned!
+ * For this reason, these methods are protected rather than public. You can choose to expose them in your database
+ * subclass if you wish, but we recommend that you instead use the typesafe, public, model-bases methods, such as
+ * {@link #update(Criterion, TableModel)}, {@link #updateWithOnConflict(Criterion, TableModel, TableStatement.ConflictAlgorithm)},
+ * {@link #delete(Class, long)}, and {@link #deleteWhere(Class, Criterion)}.
+ * <p>
+ * As a convenience, when calling the {@link #query(Class, Query) query} and {@link #fetchByQuery(Class, Query)
+ * fetchByQuery} methods, if the <code>query</code> argument does not have a FROM clause, the table or view to select
+ * from will be inferred from the provided <code>modelClass</code> argument (if possible). This allows for invocations
+ * where {@link Query#from(com.yahoo.squidb.sql.SqlTable) Query.from} is never explicitly called:
+ *
+ * <pre>
+ * SquidCursor&lt;Person&gt; cursor =
+ *         db.query(Person.class, Query.select().orderBy(Person.NAME.asc()));
+ * </pre>
+ *
+ * By convention, the <code>fetch...</code> methods return a single model instance corresponding to the first record
+ * found, or null if no records are found for that particular form of fetch.
  */
-public abstract class AbstractDatabase {
+public abstract class SquidDatabase {
 
     /**
      * @return the database name
      */
-    protected abstract String getName();
+    public abstract String getName();
```

Comment:
I don't see any harm in it being public, was more wondering if something in the codebase required it.